pageserver/src/tenant/timeline.rs

```diff
@@ -4063,17 +4063,11 @@ impl Timeline {
 
     /// Return true if the value changed
     ///
-    /// This function must only be used from the layer flush task, and may not be called concurrently.
+    /// This function must only be used from the layer flush task.
     fn set_disk_consistent_lsn(&self, new_value: Lsn) -> bool {
-        // We do a simple load/store cycle: that's why this function isn't safe for concurrent use.
-        let old_value = self.disk_consistent_lsn.load();
-        if new_value != old_value {
-            assert!(new_value >= old_value);
-            self.disk_consistent_lsn.store(new_value);
-            true
-        } else {
-            false
-        }
+        let old_value = self.disk_consistent_lsn.fetch_max(new_value);
+        assert!(new_value >= old_value);
```

Comment:
could you maybe add a message about the disk consistent lsn being required to always increase?

Reply:
not sure actually :) I will at least include the requirement in text and the values.

the reason is:
- at runtime a timeline can only ingest later Lsn
- we can go backwards with disk_consistent_lsn only when timeline is timeline is loaded

but I cannot find a way to formulate that into a single sentence.

Reply:
Maybe "while attached" is a better term than "at runtime", but your text is fine as well. Feel free to resolve this.